internal/driver/driver.go

```diff
@@ -0,0 +1,792 @@
+func (s *Driver) HandleReadCommands(deviceName string, protocols map[string]models.ProtocolProperties, reqs []sdkModel.CommandRequest) (res []*sdkModel.CommandValue, err error) {
+	s.lc.Debugf("Driver.HandleReadCommands: protocols: %v, resource: %v, attributes: %v", protocols, reqs[0].DeviceResourceName, reqs[0].Attributes)
+
+	// assume the max batch size is 16, must be less than 20
+	var batch_size = 16
+
+	var reqs_len = len(reqs)
+	var s7DataItems = []gos7.S7DataItem{}
+
+	res = make([]*sdkModel.CommandValue, reqs_len)
+	var s7_errors = make([]string, reqs_len)
+
+	// two-dimensional array for handle S7DataItems
+	var dataset = make([][]byte, reqs_len)
+	for i := range dataset {
+		dataset[i] = make([]byte, 4) // 4 bytes
+	}
+
+	// Get S7 device connection information, each Device has its own connection.
+	s7Client := s.getS7Client(deviceName, protocols)
+
+	// assemble s7DataItems, get items, fetch data
+
+	var times = int(reqs_len / batch_size)
+	var remains = reqs_len % batch_size
+	var tmp_reqs []sdkModel.CommandRequest
+
+outloop:
+	for j := 0; j <= times; j++ {
+
+		// 1. init array
+		s7DataItems = s7DataItems[:0] // clear array
+		tmp_reqs = tmp_reqs[:0]       // clear array
+		if j >= times {
+			if remains > 0 {
+				tmp_reqs = reqs[times*batch_size : times*batch_size+remains]
+			} else {
+				break outloop // load complete, exit loop
+			}
+		} else {
+			tmp_reqs = reqs[j*batch_size : j*batch_size+batch_size]
+		}
+
+		// 2. get resources from reqs append to items
+		for i, req := range tmp_reqs {
+
+			nodename := cast.ToString(req.Attributes["NodeName"])
+			dbInfo, _ := s.getDBInfo(nodename)
+
+			var s7DataItem = gos7.S7DataItem{
+				Area:     dbInfo.Area,
+				WordLen:  dbInfo.WordLength,
+				DBNumber: dbInfo.DBNumber,
+				Start:    dbInfo.Start,
+				Amount:   dbInfo.Amount,
+				Data:     dataset[j*batch_size+i],
+				Error:    s7_errors[i],
+			}
+			s7DataItems = append(s7DataItems, s7DataItem)
+		}
+		s.lc.Debugf("Read from S7DataItems: ", s7DataItems)
+
+		// 3. use AGReadMulti api to get values from S7 device, if error, try 3 times
+		retrytimes := 3
+		for {
+			err = s7Client.Client.AGReadMulti(s7DataItems, len(s7DataItems))
+			if err != nil {
+				s.lc.Errorf("AGReadMulti Error: %s, reconnecting...", err)
+				s.mu.Lock()
+				s.s7Clients[deviceName] = nil
+				s.mu.Unlock()
+				s.s7Clients[deviceName] = s.NewS7Client(deviceName, protocols)
+				s7Client = s.s7Clients[deviceName]
+			} else {
+				s.lc.Debugf("AGReadMulti read from 'dataset': ", dataset)
+				break
+			}
+
+			retrytimes--
+			if retrytimes == 0 {
+				break
+			}
+
+		}
+
+	}
+	// end assemble s7DataItems
+
+	for _, s7_error := range s7_errors {
+		if s7_error != "" {
+			s.lc.Errorf("S7 Client AGReadMulti error: %s", s7_error)
+			return
+		}
+	}
+
+	// read results from the dataset of s7DataItems
+	for i, req := range reqs {
+
+		var result = &sdkModel.CommandValue{}
+		var value interface{}
```

Comment:
replace all `interface{}` with the keyword `any`
this is the current golang coding style